bot.py

```diff
@@ -85,6 +84,17 @@ async def off_command(update: Update, context: ContextTypes.DEFAULT_TYPE):
     else:
         await update.message.reply_text("You don't have an active session. Use /on to start the timer.")
 
+async def abort_command(update: Update, context: ContextTypes.DEFAULT_TYPE):
+    user_name = update.effective_user.username
+    if user_name in user_sessions:
+        if user_name == "Unknown":
+            await update.message.reply_text("You are not registered yet. Contact @samtjong to register before you can use this bot.")
```

Comment:
Not needed. Check for unknown user only in 1 place. Dont need check here.

Reply:
For now, `if` gate is implemented against unknown users at the command-level.

Might consider checking for unknown users early on to avoid repeating `if` gates, i.e. somewhere in `main()` instead of repeating in `on_command()` and, in the future, `hour_command()`.

May open a new PR for this.